provider/testdata/Cargo.toml

```diff
@@ -299,65 +300,73 @@ displaydoc = { version = "0.2.3", default-features = false, optional = true }
 writeable = { version = "0.4", path = "../../utils/writeable", optional = true }
 
 # databake deps
-icu_calendar = { version = "1.0.0-beta1", path = "../../components/calendar", optional = true }
-icu_casemapping = { version = "0.2", path = "../../experimental/casemapping", optional = true }
-icu_collator = { version = "1.0.0-beta1", path = "../../components/collator", optional = true }
-icu_datetime = { version = "1.0.0-beta1", path = "../../components/datetime", features = ["experimental"], optional = true }
-icu_decimal = { version = "1.0.0-beta1", path = "../../components/decimal", optional = true }
-icu_list = { version = "1.0.0-beta1", path = "../../components/list", optional = true }
-icu_locid_transform = { version = "1.0.0-beta1", path = "../../components/locid_transform", optional = true }
-icu_normalizer = { version = "1.0.0-beta1", path = "../../components/normalizer", optional = true }
-icu_plurals = { version = "1.0.0-beta1", path = "../../components/plurals", optional = true }
-icu_properties = { version = "1.0.0-beta1", path = "../../components/properties", optional = true }
-icu_segmenter = { version = "1.0.0-alpha1", path = "../../experimental/segmenter", optional = true, features = ["lstm"] }
-icu_timezone = { version = "1.0.0-beta1", path = "../../components/timezone", optional = true }
-icu_collections = { version = "1.0.0-beta1", path = "../../components/collections", optional = true }
-litemap = { version = "0.5", path = "../../utils/litemap", optional = true, default-features = false }
-tinystr = { version = "0.6", path = "../../utils/tinystr", optional = true }
-zerovec = { version = "0.8", path = "../../utils/zerovec", optional = true }
+icu_locid = { version = "1.0.0-beta1", path = "../../components/locid" }
+icu_collections = { version = "1.0.0-beta1", path = "../../components/collections" }
+litemap = { version = "0.5", path = "../../utils/litemap", default-features = false }
+tinystr = { version = "0.6", path = "../../utils/tinystr" }
+zerovec = { version = "0.8", path = "../../utils/zerovec" }
 
-[dev-dependencies]
-icu_plurals = { version = "1.0.0-beta1", path = "../../components/plurals" }
+# databake options
+icu_calendar = { version = "1.0.0-beta1", path = "../../components/calendar", default-features = false, optional = true }
+icu_casemapping = { version = "0.2", path = "../../experimental/casemapping", default-features = false, optional = true }
+icu_collator = { version = "1.0.0-beta1", path = "../../components/collator", default-features = false, optional = true }
+icu_datetime = { version = "1.0.0-beta1", path = "../../components/datetime", features = ["experimental"], default-features = false, optional = true }
```

Comment:
Issue: As discussed previously, I do not want `icu_testdata/icu_datetime` to enable `icu_datetime/experimental`. It makes it much harder to reach the non-experimental version of the DateFormatter and DateTimeFormatter constructors, which has the following implications:

- Impossible to reach these code paths in tests (docs, unit, or integration)
- Clients using icu_testdata pull in the skeletons data and code and don't get compile errors for trying to use it, making it more risky when we try changing those APIs post 1.0

I would rather you introduce a feature `icu_datetime_experimental` instead.

Reply:
+1

Reply:
IIRC, one of the challenges here was adding cfgs to BakedDataProvider to selectively exclude parts of a crate's data.

Suggestion: Disable `experimental` features when generating testdata's BakedDataProvider and rely on postcard for testing experimental features (either only in datetime or across the board).

Reply:
I wanted to do this in a follow up but ok.

I think it's really bad that the `icu_datetime/experimental` feature disables the non-experimental constructor. Features are meant to be additive and non-additivity gets us into situations like this. Why don't we give the experimental constructor a different name while it's experimental?

Reply:
But why?

Reply:
Introducing this feature breaks two examples that are using `Typed[Zoned]DateTimeFormatter`, something activates `icu_datetime/experimental` (don't see what or how) so the constructors require the skeleton bound, but that's not available on `icu_testdata` anymore.

Reply:
#2495 should maybe fix the examples issue.

As discussed, please have separate features for the experimental crates